source-code/core/messages/src/utilities.ts

```diff
@@ -0,0 +1,224 @@
+
+// @ts-nocheck
+import type { LanguageTag } from "@inlang/language-tag"
+import type { Message, Variant } from "./schema.js"
+import type { Result } from "@inlang/result"
+import { MessagePatternsForLanguageTagDoNotExistError, MessageVariantAlreadyExistsError, MessageVariantDoesNotExistError } from "./errors.js"
+
+/**
+ * Get the variant of a message
+ *
+ * All actions are immutable.
+ *
+ * @example
+ * 	const variant = getVariant(message, { languageTag: "en", selectors: { gender: "male" }});
+ */
+export function getVariant(
+	message: Message,
+	options: {
+		languageTag: LanguageTag
+		selectors?: Record<string, string>
+	},
```

Comment:
@NilsJacobsen this API is unexpected. I would have expected:

```ts
const variant = getVariant(message.body[languageTag], selectors: { gender: "male" })
```

- getting the language tag is straightforward via the body
- a variant is unrelated to the language tag (this api seems like a leaky design)
- variants only concern a specific message body

Reply:
Also unexpected that the function returns a `Result`. The function can't crash. Either a variant exists or it doesn't (undefined). 

I will go ahead and change the API because it's obvious that we will change the API and I have to work with getVariant at the moment. Don't want to refactor it again. 

@NilsJacobsen use `Result` for failing code, not for code that can return `undefined`.

Reply:
NVM the variants need the message selectors to derive the ordering of the variants. Hence, passing the entire message is correct but unexpected. The return type of result is incorrect. I will change that.